events/views.py

```diff
@@ -1,44 +1,36 @@
-# TODO: clean imports
-from rest_framework import status
-from rest_framework.decorators import api_view, permission_classes
-from rest_framework.permissions import IsAdminUser
+from rest_framework.decorators import api_view
 from rest_framework.response import Response
 from rest_framework.generics import ListAPIView
-from rest_framework.viewsets import ModelViewSet
 from .models import Event, EventRegister
 from .serializers import *
-from decorators import ecell_user,relax_ecell_user
-from django.http import HttpResponse
-from django.views.decorators.csrf import csrf_exempt
-# from django.utils.six.moves.urllib.parse import urlsplit
-import csv
-
-
-# TODO: simplify with drf
-@api_view(['GET', ])
-def get_events(request, year):
-    # print(request.META['SERVER_PROTOCOL'])
-    res_message = ""
-    res_status = ""
-    res_data = []
-    # scheme = urlsplit(request.build_absolute_uri(None)).scheme
-    # print(scheme)
-    events = Event.objects.filter(year=year, flag=True)
-    if len(events) > 0:
-        res_data = EventListSerializer(
-            events, many=True, context={
-                'request': request}).data
-        res_message = "Events Fetched successfully."
-        res_status = status.HTTP_200_OK
-    else:
-        res_message = "Events Couldn't be fetched"
-        res_status = status.HTTP_404_NOT_FOUND
-
-    return Response({
-        "message": res_message,
-        "data": res_data
-    }, status=res_status)
+from decorators import ecell_user
+from rest_framework import status, generics
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework.permissions import AllowAny
+from utils.swagger import set_example
+from . import responses
 
+class EventView(generics.ListAPIView):
+    permission_classes = [AllowAny, ]
+    queryset = Event.objects.all()
+    serializer_class = EventListSerializer
+    @swagger_auto_schema(
+        operation_id='get_events',
+        request_body=EventListSerializer,
```

Comment:
```suggestion
```
`get` requests do not have a `request_body`.

Reply:
done ma'am